src/napari/_vispy/overlays/base.py

```diff
@@ -62,67 +60,17 @@ class VispyCanvasOverlay(VispyBaseOverlay):
     def __init__(self, *, overlay, node, parent=None) -> None:
         super().__init__(overlay=overlay, node=node, parent=parent)
 
-        # offsets and size are used to control fine positioning, and will depend
+        # size is used to control fine positioning, and will depend
         # on the subclass and visual that needs to be rendered
-        self.x_offset = 10.0
-        self.y_offset = 10.0
         self.x_size = 0.0
         self.y_size = 0.0
         self.node.transform = STTransform()
         self.overlay.events.position.connect(self._on_position_change)
+        self.canvas_position_callback = lambda: None
 
     def _on_position_change(self, event=None):
-        # subclasses should set sizes correctly and adjust offsets to get
-        # the optimal positioning
-        if self.node.parent is None:
-            return
-        x_max, y_max = list(self.node.parent.size)
-        position = self.overlay.position
-
-        if position == CanvasPosition.TOP_LEFT:
-            transform = [self.x_offset, self.y_offset, 0, 0]
-        elif position == CanvasPosition.TOP_CENTER:
-            transform = [x_max / 2 - self.x_size / 2, self.y_offset, 0, 0]
-        elif position == CanvasPosition.TOP_RIGHT:
-            transform = [
-                x_max - self.x_size - self.x_offset,
-                self.y_offset,
-                0,
-                0,
-            ]
-        elif position == CanvasPosition.BOTTOM_LEFT:
-            transform = [
-                self.x_offset,
-                y_max - self.y_size - self.y_offset,
-                0,
-                0,
-            ]
-        elif position == CanvasPosition.BOTTOM_CENTER:
-            transform = [
-                x_max / 2 - self.x_size / 2,
-                y_max - self.y_size - self.y_offset,
-                0,
-                0,
-            ]
-        elif position == CanvasPosition.BOTTOM_RIGHT:
-            transform = [
-                x_max - self.x_size - self.x_offset,
-                y_max - self.y_size - self.y_offset,
-                0,
-                0,
-            ]
-        else:
-            raise ValueError(
-                trans._(
-                    'Position {position} not recognized.',
-                    deferred=True,
-                    position=position,
-                )
-            )
-
-        self.node.transform.translate = transform
-        scale = abs(self.node.transform.scale[0])
-        self.node.transform.scale = [scale, 1, 1, 1]
+        # NOTE: subclasses should set sizes correctly to get correct positioning!
```

Comment:
It would be helpful to call this out in the docstring for the class. Does the note refer to subclasses of `VispyCanvasOverlay` or `VispyBaseOverlay`?

Reply:
Added. It's for `VispyCanvasOverlay` specifically.